catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroSettingsActivityTest.java

```diff
@@ -0,0 +1,151 @@
+import android.preference.PreferenceManager;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.espresso.intent.Intents;
+import android.support.test.espresso.matcher.PreferenceMatchers;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.SettingsActivity;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.hamcrest.Matcher;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.intent.Intents.intended;
+import static android.support.test.espresso.intent.Intents.intending;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.hasAction;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.hasData;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.ui.BaseSettingsActivity.SETTINGS_SHOW_PHIRO_BRICKS;
+import static org.catrobat.catroid.ui.SettingsActivity.PHIRO_INITIALIZED;
+import static org.hamcrest.Matchers.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroSettingsActivityTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<SettingsActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(SettingsActivity.class, true, false);
+
+	private List<String> allSettings = new ArrayList<>(Arrays.asList(SETTINGS_SHOW_PHIRO_BRICKS, PHIRO_INITIALIZED));
+	private Map<String, Boolean> initialSettings = new HashMap<>();
+
+	@Before
+	public void setUp() {
+		SharedPreferences sharedPreferences = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		for (String setting : allSettings) {
+			initialSettings.put(setting, sharedPreferences.getBoolean(setting, false));
+		}
+		setAllSettingsTo(false);
+
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void phiroIsInitiallyActivatedTest() {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+		assertTrue("Phiro preference was not initially enabled!", sharedPreferences.getBoolean(SETTINGS_SHOW_PHIRO_BRICKS, false));
+	}
+
+	@Test
+	public void phiroIsFirstVisibleSettingTest() {
+		onData(PreferenceMatchers.withTitle(R.string.preference_title_enable_phiro_bricks))
+				.atPosition(0)
```

Comment:
This is necessary here because I want to explicitly check whether this preference is the first in the list. Without .atPosition(0) the test would pass, but shouldn't!